pydy/tests/test_joints.py

```diff
@@ -0,0 +1,275 @@
+        assert self.pinjoint.child_joint_piont == point2
+        assert self.pinjoint.parent_joint_vector == self.parent.frame.x
+        assert self.pinjoint.child_joint_vector == self.child.frame.y
+
+    def test_slidingjoint_functions(self):
+        self.slidingjoint._set_parent_child_rel()
+        assert self.parent.child == self.child
+        assert self.child.parent == self.parent
+
+        self.slidingjoint._locate_joint_point()
+        child_masscenter = self.child.masscenter
+        parent_masscenter = self.parent.masscenter
+        parent_point = parent_masscenter.locatenew(self.name + '_Point', self.parent_joint_vector)
+        child_point = child_masscenter.locatenew(self.name + '_Point', self.child_joint_vector)
+        assert self.pinjoint.parent_joint_vector == parent_point
+        assert self.pinjoint.child_joint_vector == child_point
+
+        self.join_frames()
+        assert cross(self.pinjoint.parent_axis, self.pinjoint.child_axis) == self.child.frame.z
+        assert cross(self.pinjoint.child_axis, self.pinjoint.parent_axis) == - self.parent.frame.z
+
+        self.align_axes()
+        parent_axis = self.pinjoint.parent_axis
+        child_axis = self.pinjoint.child_axis
+        assert acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())) == 0
+
+        # Similar to Pinjoint but have a velocity instead of angular velocity.
+        assert self.child_point.vel(self.parent.frame) != 0
+
+class TestCylindricalJoint():
+    # Note all the functionalitites are similar to other PinJoint and SlidingJoint.Only CylindricalJoint's specific
+    # tests are written.
+    def setup(self):
+        self.parent = Body('parent')
+        self.child = Body('child')
+        self.cylindricaljoint = CylindricalJoint('cylindricaljoint', self.parent, self.child, par_point_vec_tuple=(1,1,1),
+                                                 child_point_vec_tuple=(0,0,1), parent_direction='x + 2*y', child_direction = 'z')
+                                                 #Similar approach can be used in other joints as well.
+
+    def test_cylindricaljoint_parameters_assignment(self):
+        assert self.cylindricaljoint.parent == self.parent
+        assert self.cylindricaljoint.child == self.child
+        assert self.cylindricaljoint.parent_joint_vector == self.parent.frame.x + self.parent.frame.y + self.parent.frame.z
+        assert self.cylindricaljoint.child_joint_vector == self.child.frame.z
+        assert self.cylindricaljoint.parent_direction == self.parent.frame.x + 2*self.parent.frame.y
+        assert self.cylindricaljoint.child_direction == self.child.frame.z
+
+    def test_cylindricaljoint_apply_joint(self):
+        assert self.child_point.vel(self.parent.frame) != 0
+        parent_axis = self.pinjoint.parent_direction
+        child_axis = self.pinjoint.child_direction
+        assert acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())) == 0
+
```

Comment:
This test is missing all info about the two coordinates and the two velocities associated with this joint.